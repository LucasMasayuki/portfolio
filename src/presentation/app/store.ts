/* eslint-disable import/no-cycle */
import { Action, configureStore, ThunkAction } from '@reduxjs/toolkit';
import { colorModeSlice } from '../stores/color-mode-slice';
import { notificationSlice } from '../stores/notification-slice';

export const store = configureStore({
  reducer: {
    colorMode: colorModeSlice.reducer,
    notification: notificationSlice.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
