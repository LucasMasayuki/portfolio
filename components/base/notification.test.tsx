import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { describe, expect, it } from 'vitest';
import { store } from '../../src/presentation/app/store';
import {
  AlertSeverity,
  openNotification,
} from '../../src/presentation/stores/notification-slice';
import Notification from './notification';

describe.concurrent('Notification component', () => {
  describe('when first loaded', () => {
    it('should not open snackbar', () => {
      const { container } = render(
        <Provider store={store}>
          <Notification />
        </Provider>
      );

      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when change notification state', () => {
    it('should open snackbar, when is open is true', () => {
      const { container } = render(
        <Provider store={store}>
          <Notification />
        </Provider>
      );

      store.dispatch(
        openNotification({
          message: 'test',
          severity: AlertSeverity.ERROR,
        })
      );

      expect(container.innerHTML).toMatchSnapshot();
    });
  });
});
