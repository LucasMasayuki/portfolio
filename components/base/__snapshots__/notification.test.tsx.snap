// Vitest Snapshot v1

exports[`Notification component > when change notification state > should open snackbar, when is open is true 1`] = `""`;

exports[`Notification component > when first loaded > should not open snackbar 1`] = `""`;
